### PR TITLE
user and uuid conversion

### PR DESCRIPTION
**Pull request by Zickles**
# Change Log
- Add user to uuid util
- Add uuid to user util

### CODE REVIEW

**Comment by Ascynx** [comment]
I believe you should use a cache to avoid calling the mojang api for duplicates. Could also be populated with other players in the same lobby.

**Comment by Ascynx** [comment]
Also might be a good idea to run http requests on another thread to avoid it causing the game "freezing"

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class APIUtils {
+
+    private static JsonElement PlayerInfo;
+    private static String Player;
+
+    private static HashMap<String, String> PlayerData = new HashMap<String, String>();
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: What's the naming convention with these variables doing?

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        if (PlayerData.containsValue(uuid)) {
+            for(Map.Entry<String, String> entry : PlayerData.entrySet()){
+                if(Objects.equals(entry.getValue(), uuid)) {
+                    Player = entry.getKey();
+                }
+            }
+            System.out.println("Using cached player data...");
+            return Player;
+        }
+        else {
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: This else is completely unnecessary.

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                PlayerData.put(PlayerInfo.getAsJsonObject().get("name").getAsString(), PlayerInfo.getAsJsonObject().get("id").getAsString());
+
+                in.close();
+            } catch(Exception e) {
+                System.out.println(e);
+            }
+            return PlayerInfo.getAsJsonObject().get("name").getAsString();
+        }
+    }
+    public static String getUUID(String username) {
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: Can we get some spacing (newlines) in here? Very hard to read.

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        if (PlayerData.containsKey(username)) {
+            for(Map.Entry<String, String> entry : PlayerData.entrySet()){
+                if(Objects.equals(entry.getKey(), username)) {
+                    Player = entry.getValue();
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: Why are we setting the static variable "Player" for the whole class whenever a single player is looked up?

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        else {
+            try {
+                URL userinfo = new URL("https://sessionserver.mojang.com/session/minecraft/profile/" + uuid);
+                URLConnection urlConnect = userinfo.openConnection();
+                BufferedReader in = new BufferedReader(new InputStreamReader(urlConnect.getInputStream()));
+                PlayerInfo = JsonParser.parseReader(in);
+
+                PlayerData.put(PlayerInfo.getAsJsonObject().get("name").getAsString(), PlayerInfo.getAsJsonObject().get("id").getAsString());
+
+                in.close();
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: Could this not be a try-with-resources?

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            } catch(Exception e) {
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: This catch is really broad, please narrow it down

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+    private static HashMap<String, String> PlayerData = new HashMap<String, String>();
+    public static String getUsername(String uuid) {
+        if (PlayerData.containsValue(uuid)) {
+            for(Map.Entry<String, String> entry : PlayerData.entrySet()){
+                if(Objects.equals(entry.getValue(), uuid)) {
+                    Player = entry.getKey();
+                }
+            }
+            System.out.println("Using cached player data...");
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: Please use the loggers.

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    private static String Player;
+
+    private static HashMap<String, String> PlayerData = new HashMap<String, String>();
+    public static String getUsername(String uuid) {
+        if (PlayerData.containsValue(uuid)) {
+            for(Map.Entry<String, String> entry : PlayerData.entrySet()){
+                if(Objects.equals(entry.getValue(), uuid)) {
+                    Player = entry.getKey();
+                }
+            }
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: [Please use HashMap#get](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#get-java.lang.Object-)

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class APIUtils {
+
+    private static JsonElement PlayerInfo;
+    private static String Player;
```
Comment (TGWaffles) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: Why are these class fields?

**Review by TGWaffles — DISMISSED** [review]

File: src/main/java/io/github/skyblockcore/util/APIUtils.java (hunk 1 above)
Comment (Zickles) [reply src/main/java/io/github/skyblockcore/util/APIUtils.java]: Will fix but any suggestions I can't think of anything

Comment (TGWaffles) [reply src/main/java/io/github/skyblockcore/util/APIUtils.java]: > Will fix but any suggestions I can't think of anything

https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

File: src/main/java/io/github/skyblockcore/util/APIUtils.java (hunk 8 above)
Comment (Harry282) [reply src/main/java/io/github/skyblockcore/util/APIUtils.java]: lmao

Comment (Zickles) [reply src/main/java/io/github/skyblockcore/util/APIUtils.java]: it was my 5 am brain ok? i forgot that exsits

**Comment by Zickles** [comment]
build is going to fail bc kr set import to SkyblockCore instead of SkyBlockCore
will fix soon

File: src/main/java/io/github/skyblockcore/util/APIUtils.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+            return null;
+        }
+    }
+
+    public static String getUUID(String username) {
+        Optional<Map.Entry<String, String>> cachedUuid = uuidToUsernameCache.asMap().entrySet().stream().filter(e -> e.getValue().equalsIgnoreCase(username)).findAny();
+        if (cachedUuid.isPresent()) {
+            LOGGER.info("Using cached player data...");
```
Comment (Ascynx) [inline src/main/java/io/github/skyblockcore/util/APIUtils.java]: I don't think it's needed to add this to log, could maybe be used for debugging but I think then we can use the dev mode.

Comment (Ascynx) [reply src/main/java/io/github/skyblockcore/util/APIUtils.java]: Same for line24

**Review by Ascynx — COMMENTED** [review]
Outside of that it seems fine for me.

**Review by Ascynx — APPROVED** [review]

**Review by Harry282 — APPROVED** [review]

**Pull request merged**